imports/plugins/included/product-variant/client/templates/products/productGrid/productGrid.js

```diff
@@ -15,6 +15,7 @@ Template.productGrid.onCreated(function () {
 });
 
 Template.productGrid.onRendered(function () {
+  Meteor.call("products/removeBlankProducts");
```

Comment:
Looks like you're deleting the products when rendering the page, uh?

Reply:
yes. the way reaction commerce is structured. it creates the product Id immediately you click the add product button. when you enter the changes in the field it updates the products.
So, I delete the blank products from the product grid.